tests/integration/lsig_test.py

```diff
@@ -0,0 +1,133 @@
+from itertools import product
+from pathlib import Path
+
+import pytest
+
+
+from graviton.blackbox import (
+    DryRunExecutor as Executor,
+    DryRunInspector as Inspector,
+)
+
+from tests.clients import get_algod
+
+ALGOD = get_algod()
+TESTS_DIR = Path.cwd() / "tests"
+
+
+@pytest.mark.parametrize("suffix", ("V1", "V2", "V3", "V4"))
```

Comment:
@tzaffi Apologies if I missed an explanation elsewhere, but can you help me understand the need to retain 4 versions of `lsig_factorizer_game_1_5_7_`?

Reply:
I don't think these different variants are sufficiently useful to maintain, vs. their cost. Should I only keep V4?

Reply:
@tzaffi I've not really inspected the differences.  I raised the question based on how the different versions are used in the tests.  From the test usage patterns, I was unable to understand what's unique.

Just now, I started peeking at diffs.  I see in some cases the diffs look small.

Since I'm unaware of the lineage, I'm leaning on your perspective.  Here's my 2c:
* Based on the test patterns alone, it seems retaining 4 versions isn't needed.
* If you feel there's not much uniqueness here, then I definitely favor removing other versions.
* If there is a reason to retain the 4 versions, then I appreciate a brief inline comment to help maintainers understand the variation.

----

Aside:  Additionally, I am realizing that I'm unsure how the TEAL source was generated.  How did these files generated?

Reply:
I'll remove the versions previous to 4 and remove versioning from its name.

Once again, pyteal [is the author here](https://github.com/algorand/pyteal/blob/1ee615117da7617078fa2d68ad8546966a734eb7/tests/integration/pure_logicsig_test.py#L66).

Reply:
Summarizing a verbal @tzaffi and I had:  We agreed that it'd help maintainers to add a note pointing to corresponding PyTeal sources.

Leaving thread open as a visual reminder until addressed and then ready to resolve.

Reply:
@michaeldiamant - I've added an explanatory comment at the top of [abi_test.py](https://github.com/algorand/graviton/blob/945aba2e74c7c6e07dfb6b05bf5de846b7524afa/tests/integration/abi_test.py#L2)

Let me know if you have further suggestions.

Reply:
@tzaffi I think a similar treatment for `lsig_test.py` makes sense.  I opened a child PR for your review:  https://github.com/algorand/graviton/pull/19.

Reply:
resolving as all's merged/fixed